storm-core/src/jvm/org/apache/storm/state/KeyValueState.java

```diff
@@ -45,4 +47,9 @@
      * @return the value or defaultValue if no mapping is found
      */
     V get(K key, V defaultValue);
+
+    /**
+     * @return all key value mappings as an unmodifiable map.
+     */
+    Map<K, V> getAll();
```

Comment:
You can implement custom iterator which stores cursor for `hscan` as internal state. Call hscan to get some elements and store to `queue` as internal state, pop and return it when next() is called. If queue is empty you can hscan again to determine there're more elements in the hash or it is end of elements in hash.